### PR TITLE
Add channel moderation to bookmarks

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR adds the bookmarks permissions to the channel moderation system. Sits on top of https://github.com/mattermost/mattermost/pull/25700, so it should be merged after it.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55633

#### Release Note
```release-note
Adds the channel bookmarks permissions to the channel moderation system.
```

### CODE REVIEW

**Comment by mgdelacroix** [comment]
The PR currently shows commits from the branches that it builds on top of. To review only the contents related to the PR it's better to use the commit https://github.com/mattermost/mattermost/pull/25716/commits/79145ece9821b2d509a33b2f7785c7e855f8ea59 instead of the diff view

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
@mgdelacroix Needs rebase

**Comment by mgdelacroix** [comment]
@enahum rebased!

File: server/channels/api4/channel_bookmark_test.go [hunk 1]
```diff
@@ -533,88 +586,6 @@ func TestEditChannelBookmark(t *testing.T) {
 		require.Equal(t, th.BasicUser2.Id, ucb.Updated.OwnerId)
 	})
 
-	t.Run("a user should be able to edit another usrer's bookmark", func(t *testing.T) {
```
Comment (mgdelacroix) [inline server/channels/api4/channel_bookmark_test.go:536]: I fixed a typo in one of the first rebases and this test (mentioning `usrer's`) got duplicated... twice. Removed here

File: e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.js [hunk 2]
```diff
@@ -243,6 +243,12 @@ export const enableDisableAllChannelModeratedPermissionsViaAPI = (channel, enabl
                             guests: enable,
                         },
                     },
+                    {
```
Comment (mgdelacroix) [inline e2e-tests/cypress/tests/integration/channels/enterprise/system_console/channel_moderation/helpers.js:246]: e2e tests cannot cover the full use case if we don't have a UI to check them against. I've created https://mattermost.atlassian.net/browse/MM-56198 to write them after some components of the UI are implemented

**Review by enahum — APPROVED** [review]

**Pull request merged**